app/pages/project/index.cjsx

```diff
@@ -179,10 +179,17 @@ ProjectPage = React.createClass
       @state.activeWorkflows[randomIndex].id
 
   getWorkflow: (selectedWorkflowIndex) ->
-    @setState {
-      selectedWorkflow: @state.activeWorkflows[selectedWorkflowIndex],
-      loadingSelectedWorkflow: false
-    }
+    apiClient.type('workflows').get({ id: "#{@state.activeWorkflows[selectedWorkflowIndex].id}" })
+      .catch (error) =>
+        console.error error
+        @setState {
+          loadingSelectedWorkflow: false
+        }
+      .then ([workflow]) =>
+        @setState {
```

Comment:
Should this perhaps also store the full workflow in `state.activeWorkflows[selectedIndex]`? We'd need to check whether we have the full resource already, before the API request, so that we don't request it again this session.